include/oneapi/dpl/internal/histogram_extension_defs.h

```diff
@@ -0,0 +1,48 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _ONEDPL_HISTOGRAM_EXTENSION_DEFS_H
+#define _ONEDPL_HISTOGRAM_EXTENSION_DEFS_H
+
+#include "../pstl/onedpl_config.h"
+
+namespace oneapi
+{
+namespace dpl
+{
+
+// Only supported for dpcpp backend
+#ifdef _ONEDPL_BACKEND_SYCL
+
+template <typename _ExecutionPolicy, typename _RandomAccessIterator1, typename _Size, typename _T,
+          typename _RandomAccessIterator2>
+oneapi::dpl::__internal::__enable_if_execution_policy<_ExecutionPolicy, _RandomAccessIterator2>
+histogram(_ExecutionPolicy&& policy, _RandomAccessIterator1 __first, _RandomAccessIterator1 __last,
+          const _Size& __num_bins, const _T& __first_bin_min_val, const _T& __last_bin_max_val,
```

Comment:
Usually, there is practice (according to std signatures) to pass  Size/count by value.

Reply:
1.  Sure, I'll make that change.  
2. Sure, we can do that.  I suppose its maybe better to convert to the value type here than to wait until the comparisons to reconcile types.

I'll update when this is done.

Reply:
@MikeDvorskiy, do you think we should also pass `__first_bin_[min/max]_val` by value as well? In most cases its probably better.

Reply:
I've converted to have all the scalar* inputs to be passed by value and utilized the `value_type`.  Please take a look to see if this is what you had in mind.  
I've converted `_T` to be a defaulted template parameter (`_ValueType`) which is default equal to `iterator_traits<_RandomAccessIterator1>::value_type`.